### ISSUE TITLE
In the captcha warning, place a link of the site which issued the captcha

### ISSUE THREAD

**Issue by cyclaero** [searx/searx#2672]
**Is your feature request related to a problem? Please describe.**

Recently I got a captcha warning on my locally installed searx related to Qwant. I only needed to visit https://qwant.com/ and I was again presented with the captcha. I solved the captcha, and from that point on, Qwant started to work via searx without anymore issues.

**Describe the solution you'd like**

A link or button to the site which threw the captcha, _„Please click here and solve the Captcha!“_. Background-color light-orange instead of red, since it might be a temporary failure only.

Most of us are humans who use searx, and at least on local searx instances resolving a captcha problem could be a no-brainer even for newbies, when there would be helpful hint, telling to go there and to solve this CAPTCHA!!!.

For remote searx instances, this won’t work of course.

**Describe alternatives you've considered**

Remote searx instances could pass the exact original captcha to solve in an iframe floating above the searx-page. The user could solve the captcha, and pass the result to the server, which in turn would deliver the response, and the remote server would be clean for some time.

OK, in a another window would also be possible, However, I would prefer an iframe.

Perhaps this would be the ideal solution for local searx instance as well.

**Comment by unixfox** [comment]
I'm closing this feature request because most of the CAPTCHA can't be loaded from the Searx page (iframe) due to CORS: https://en.wikipedia.org/wiki/Cross-origin_resource_sharing

At least that's how Google recaptcha, the captcha from Cloudflare and yandex captcha work. Those are the most common ones on Searx.

And for the local installation of Searx it would be too difficult to redirect to the CAPTCHA page in a generic way because for most of the CAPTCHA (maybe not qwant) the page is unique and cannot be loaded a second time.

If you want to combat CAPTCHA you will have to use proxies. I already asked in https://github.com/searx/searx/issues/1812 if we could have an implementation for using external anti captcha system but it was denied by the owner of Searx.

**Comment by cyclaero** [comment]
In the feature request, there were to topics:

***1. Describe the solution you'd like***

Link to the site which issued a captcha

and

***2. Describe alternatives you've considered***

Pass the issued captcha in an iframe, and transparently return the result to the issuer.

To be very clear, you closed this issue because of (2), the alternative which I considered as kind of a long shot, and NOT because of (1), the solution which I liked.

Note, public instances are completely irrelevant for me, these are illegal anyways: https://eur-lex.europa.eu/legal-content/en/TXT/PDF/?uri=uriserv%3AOJ.C_.2014.052.01.0010.01.ENG

So, good luck! Local instances are the way to go.

**Comment by unixfox** [comment]
Didn't I reply to both of your topics? I closed the issue because both of the two solutions are not feasible.

I replied here for the first solution:

> And for the local installation of Searx it would be too difficult to redirect to the CAPTCHA page in a generic way because for most of the CAPTCHA (maybe not qwant) the page is unique and cannot be loaded a second time.

**Comment by cyclaero** [comment]
No, you did not.

Also see the title of the issue: _"In the captcha warning, place a link of the site which issued the captcha"_

(1) _A link or button to the site which threw the captcha, „Please click here and solve the Captcha!“. Background-color light-orange instead of red, since it might be a temporary failure only. ..._

The link is of course intended to bring me to the respective site, for example Qwant, in fact completely outside of Searx, ideally with `rel="noopener noreferrer" target="_blank“`. Now, if Searx was running on localhost, then Qwant would see the same IP coming in again, and usually would present me a captcha again. Then I solve it, and after this I could continue with searxing.

Of course, this won’t help for any public searx instances, but for local instances this would help newcomers to quickly mitigate captcha issues.

**Comment by unixfox** [comment]
> (1) _A link or button to the site which threw the captcha, „Please click here and solve the Captcha!“. Background-color light-orange instead of red, since it might be a temporary failure only. ..._

What are you referring to when talking about the `site`? Is it only the domain (example: www.example.com) that threw the CAPTCHA or the full URL that threw the CAPTCHA (example: https://www.example.org/captcha1z5asa57e)?

If it's the domain that you are talking about, is the engine name in the error message not enough for you to find the domain? Example with stackoverflow:
![2021-03-21_22-58](https://user-images.githubusercontent.com/4016501/111922220-908cc380-8a90-11eb-94b6-dde4cdf456bf.png)

If it's the full URL then like I said in first comment the URL that threw the CAPTCHA is for most of the CAPTCHA system unique and will lead to nowhere if tried a second time.
Also, most of the CAPTCHA system rely on cookies for validating that you indeed solved the CAPTCHA. In this case if you solve the CAPTCHA outside of Searx then the cookies will be stored in your browser and not in Searx so Searx will still be blocked even though your browser won't be.

Qwant is probably one of the few to rely on the IP address for validating that you solved the CAPTCHA. We aren't going to spend the time to build a solution that works for one single case and doesn't work for the others CAPTCHA system that exist on the Internet.
But you are always free to submit a pull request that would only solve your issue with the CAPTCHA system of Qwant.

**Comment by cyclaero** [comment]
> What are you referring to when talking about the site? Is it only the domain (example: www.example.com) that threw the CAPTCHA or the full URL that threw the CAPTCHA (example: https://www.example.org/captcha1z5asa57e)?

I am referring to the domain only.

> If it's the domain that you are talking about, is the engine name in the error message not enough for you to find the domain? 

For me and for you YES. We would immediately know what to do. My wife would be scared by this CAPTCHA notice and never try Searx again. My suggestion would make it a little less scary, and help the normal people (98 % don’t use Linux) to do the right thing.

> Also, most of the CAPTCHA system rely on cookies for validating that you indeed solved the CAPTCHA. ...

This would be very surprising to me. Then nobody could use Google and others in private/incognito browsing mode. On my cell phone I use solely incognito mode, and I never had to solve a captcha. However, I may be wrong. In the past (without) Searx, I saw captchas only from Qwant, and only when I issued a bunch of searches via GET requests. Anyway, normal users would easily be able to repeat the search with the engine wich refused to work from inside Searx.

> But you are always free to submit a pull request that would only solve your issue with the CAPTCHA system of Qwant.

I know, I am free to do a lot of things, perhaps I submit a PR on this issue once Easter and Christmas are going to be celebrated on the same day.
